### ISSUE TITLE
Incorrect table name in Meta-Data Schema appendix [BATCH-2853]

### ISSUE THREAD

**Issue by spring-issuemaster** [spring-projects/spring-batch#761]
**[Luigi Quitadamo](https://jira.spring.io/secure/ViewProfile.jspa?name=luigi.quitadamo@gmail.com)** opened **[BATCH-2853](https://jira.spring.io/browse/BATCH-2853?redirect=false)** and commented

Index suggested in docs (https://docs.spring.io/spring-batch/docs/current/reference/html/schema-appendix.html#recommendationsForIndexingMetaDataTables) doesn't match with tables created by spring batch.

 
|Default Table Name|Where Clause|Frequency|MATCH|
|---|---|---|---|
|BATCH_JOB_INSTANCE|JOB_NAME = ? and JOB_KEY = ?|Every time a job is launched|For SqlServer this index already exists (created by constraint)|
|BATCH_JOB_EXECUTION|JOB_INSTANCE_ID = ?|Every time a job is restarted| |
|BATCH_EXECUTION_CONTEXT|EXECUTION_ID = ? and KEY_NAME = ?|On commit interval, a.k.a. chunk|This table doesn't exist|
|BATCH_STEP_EXECUTION|VERSION = ?|On commit interval, a.k.a. chunk (and at start and end of step)| |
|BATCH_STEP_EXECUTION|STEP_NAME = ? and JOB_EXECUTION_ID = ?|Before each step execution|Is it correct the order for index? JOB_EXECUTION_ID is a foreign key so probably a better index should be (JOB_EXECUTION_ID, STEP_NAME)|

 

 Furthermore, probably system needs index on BATCH_JOB_EXECUTION_PARAMS.JOB_EXECUTION_ID because it has a foreign key constraint.

 

 

---

**Affects:** 3.0.10, 4.2.0, 4.1.1

**Referenced from:** pull request https://github.com/spring-projects/spring-batch/pull/748

**Comment by benas** [comment]
> Index suggested in docs (...) doesn't match with tables created by spring batch.

There are no indexes suggested in this section, on the contrary, it mentions that DDL scripts do not include indexes:

```
Index declarations are not included in that DDL,  because there are too many
variations in how users may want to index, [..].
The following provides **some indication** [..], so that individual projects
can make up their own minds about indexing
```

So based on this statement, I would leave any decision about indexing to the user.

That said, I agree on the following:

> BATCH_EXECUTION_CONTEXT |  This table doesn't exist

Indeed, this table does not exist anymore (it has been removed in https://github.com/spring-projects/spring-batch/commit/d272105479896ea543ce971fbb0298809b450df2#diff-9df3d792ac9c1d09e598d04018ee1f6eL71-L80). So the documentation should be updated accordingly (there is a PR for that, see #748).

**Comment by benas** [comment]
Fixed in #748 and backported to 4.2.2 and 4.1.4.
